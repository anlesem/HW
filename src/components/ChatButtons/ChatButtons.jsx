import style from './ChatButtons.module.scss';

import Stack from '@mui/material/Stack';
import Button from '@mui/material/Button';
import AddIcon from '@mui/icons-material/Add';
import DeleteIcon from '@mui/icons-material/Delete';

export const ChatButtons = ({ data }) => {
  return (
    <Stack direction="row" spacing={2} className={style.manage}>
      <Button
        variant="outlined"
        startIcon={<DeleteIcon />}
        disabled={data.checked.length > 1 ? false : true}
        data-testid="chats-button-remove"
        onClick={() => data.handleDelete(data.checked)}>
        Удалить
      </Button>
      <Button
        variant="outlined"
        startIcon={<AddIcon />}
        data-testid="chats-button-add"
        onClick={() => data.handleAdd(data.checked)}>
        Добавить
      </Button>
    </Stack>
  );
};
