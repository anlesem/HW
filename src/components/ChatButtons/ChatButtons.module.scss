.manage {
  width: 100%;
  background-color: #fff;

  display: flex;
  justify-content: space-between;
}
