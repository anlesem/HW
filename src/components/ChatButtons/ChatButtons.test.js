import { render, screen, fireEvent } from '@testing-library/react';
import { ChatButtons } from './ChatButtons';

describe('ChatButtons', () => {
  it('Компонент существует', () => {
    expect(ChatButtons).toBeInstanceOf(Function);
  });

  it('Снимок состояния', () => {
    const handleAdd = jest.fn();
    const handleDelete = jest.fn();
    const checked = [];
    const { asFragment } = render(<ChatButtons data={{ handleAdd, handleDelete, checked }} />);
    expect(asFragment(<ChatButtons />)).toMatchSnapshot();
  });

  it('Обработка клика на кнопку Добавить', () => {
    const handleAdd = jest.fn();
    const handleDelete = jest.fn();
    const checked = [];

    render(<ChatButtons data={{ handleAdd, handleDelete, checked }} />);

    const button = screen.getByTestId('chats-button-add');

    fireEvent.click(button);
    expect(handleAdd).toBeCalled();
  });

  it('Обработка клика на кнопку Удалить', () => {
    const handleAdd = jest.fn();
    const handleDelete = jest.fn();
    const checked = [0, 1];

    render(<ChatButtons data={{ handleAdd, handleDelete, checked }} />);

    const button = screen.getByTestId('chats-button-remove');

    fireEvent.click(button);
    expect(handleDelete).toBeCalled();
  });
});
