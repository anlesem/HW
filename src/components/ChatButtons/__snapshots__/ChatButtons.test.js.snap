// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatButtons Снимок состояния 1`] = `
<DocumentFragment>
  <div
    class="manage css-e53awj-MuiStack-root"
  >
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      data-testid="chats-button-remove"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
      Удалить
    </button>
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      data-testid="chats-button-add"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      Добавить
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
